### PR TITLE
feat(SFINT-5504): new methods to expand and collapse generated answer added to search and insight client

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-5504](https://coveord.atlassian.net/browse/SFINT-5504)

new analytics methods added to the search and insight clients:

- logExpandGeneratedAnswer.
- logCollapseGeneratedAnswer.

These new events will be logged when the generated answer will expanded and collapsed according to the new feature that will be supported by the RGA component  allowing the end user to expand or collapse a generated answer.

[SFINT-5504]: https://coveord.atlassian.net/browse/SFINT-5504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jelmedini — APPROVED** [review]
looks good !!!

File: src/searchPage/searchPageEvents.ts [hunk 1]
```diff
@@ -330,6 +330,14 @@ export enum SearchPageEvents {
      * Identifies the custom event that gets logged when a user activates the genQA feature.
      */
     generatedAnswerShowAnswers = 'generatedAnswerShowAnswers',
+    /**
+     * Identifies the custom event that gets logged when a user expand a generated answer.
+     */
+    expandGeneratedAnswer = 'expandGeneratedAnswer',
```
Comment (erocheleau) [inline src/searchPage/searchPageEvents.ts:336]: I'm bothered by the fact that we don't have a single "naming convention" event inside the generatedAnswer events...

Some start with `generatedAnswer` some end with `generatedAnswer` xD

Comment (erocheleau) [reply src/searchPage/searchPageEvents.ts:336]: The dev in me would prefer `generatedAnswerExpand` and `generatedAnswerCollapse` just so for intellisense and autocomplete we can easily see all generated answer events with `generatedAnswer*` but I think it's too late to request that so name them what you want :P

File: src/searchPage/searchPageClient.ts [hunk 2]
```diff
@@ -951,6 +951,22 @@ export class CoveoSearchPageClient {
         });
     }
 
+    public makeExpandGeneratedAnswer(metadata: GeneratedAnswerBaseMeta) {
+        return this.makeCustomEvent(SearchPageEvents.expandGeneratedAnswer, metadata);
+    }
+
+    public async logExpandGeneratedAnswer(metadata: GeneratedAnswerBaseMeta) {
+        return (await this.makeExpandGeneratedAnswer(metadata)).log({searchUID: this.provider.getSearchUID()});
```
Comment (erocheleau) [inline src/searchPage/searchPageClient.ts:959]: Just to be clear so the payload of these events will be:

```
{
    generativeQuestionAnsweringId: streamId,
    searchUID: lastSearchUID,
}
```

right? Plus whatever additional metadata is added on a custom event automatically?

**Review by erocheleau — APPROVED** [review]
lgtm, just questions that I have but just to confirm, everything looks good

File: src/searchPage/searchPageClient.ts (hunk 2 above)
Comment (mmitiche) [reply src/searchPage/searchPageClient.ts:959]: More precisely, that's what the payload of these new custom events will look like:
Screenshot 2024-04-18 at 4 06 30 PM

File: src/searchPage/searchPageEvents.ts (hunk 1 above)
Comment (mmitiche) [reply src/searchPage/searchPageEvents.ts:336]: Indeed, we can't fix this now and luckily we are doing a better job with the new event protocol.
 But yeah I will follow your suggestion and go with  `generatedAnswer*`

File: src/searchPage/searchPageClient.ts (hunk 2 above)
Comment (erocheleau) [reply src/searchPage/searchPageClient.ts:959]: This looks perfect to me! 👍

**Review by felixpelletier — APPROVED** [review]

**Pull request merged**